wgsl/index.bs

```diff
@@ -2703,6 +2672,83 @@ Issue: (dneto): Bitwise-complement is under discussion.  https://github.com/gpuw
   <tr><td>*e1* : *T*<br>*e2* : *T*<br>*T* is *BoolVec*<td>`e1 & e2` : *T*<td>Component-wise logical "and"
 </table>
 
+# Built-in functions # {#builtin-functions}
+
+Certain functions are always available in a [SHORTNAME] program,
+and are provided by the implementation.
+These are called *built-in functions*.
+
+Since a built-in function is always in scope, it is an error to attempt to redefine
+one or to use the name of a built-in function as an identifier for any other
+kind of declaration.
+
+Unlike ordinary functions defined in a [SHORTNAME] program,
+a built-in function may use the same function name with different
+sets of parameters.
+In other words, a built-in function may have more than one *overload*,
+but ordinary function definitions in [SHORTNAME] may not.
+
+When calling a built-in function, all arguments to the function are evaluated
```

Comment:
Wouldn't this belong to a more general "argument evaluation order" kind of thing? Should we say anything specific for builtin functions in this regard?

Reply:
Yes, generally.
That section has not yet been written for general functions.

I mentioned it specially here because in other languages (like LLVM) the behaviour of the thing corresponding to "select"  could be special. It could behave as if it is short-circuiting with respect to passing on poison values.  See the discussion in https://www.cs.utah.edu/~regehr/papers/undef-pldi17.pdf  especially section 3.4.